lib/nfc/protocols/st25tb/st25tb_poller_i.c

```diff
@@ -0,0 +1,300 @@
+    do {
+        ret = st25tb_poller_frame_exchange(
+            instance, instance->tx_buffer, instance->rx_buffer, ST25TB_FDT_FC);
+        if(ret != St25tbErrorNone) {
+            break;
+        }
+
+        if(bit_buffer_get_size_bytes(instance->rx_buffer) != 1) {
+            FURI_LOG_D(TAG, "Unexpected Initiate response size");
+            ret = St25tbErrorCommunication;
+            break;
+        }
+        if(chip_id) {
+            *chip_id = bit_buffer_get_byte(instance->rx_buffer, 0);
+        }
+    } while(false);
+
+    return ret;
+}
+
+St25tbError st25tb_poller_async_activate(St25tbPoller* instance, St25tbData* data) {
+    furi_assert(instance);
+    furi_assert(instance->nfc);
+
+    st25tb_reset(data);
+
+    St25tbError ret;
+
+    do {
+        ret = st25tb_poller_async_initiate(instance, &data->chip_id);
+        if(ret != St25tbErrorNone) {
+            break;
+        }
+
+        instance->state = St25tbPollerStateActivationInProgress;
+
+        bit_buffer_reset(instance->tx_buffer);
+        bit_buffer_reset(instance->rx_buffer);
+
+        // Send Select(Chip_ID), let's just assume that collisions won't ever happen :D
+        bit_buffer_append_byte(instance->tx_buffer, 0x0E);
+        bit_buffer_append_byte(instance->tx_buffer, data->chip_id);
+
+        ret = st25tb_poller_frame_exchange(
+            instance, instance->tx_buffer, instance->rx_buffer, ST25TB_FDT_FC);
+        if(ret != St25tbErrorNone) {
+            instance->state = St25tbPollerStateActivationFailed;
+            break;
+        }
+
+        if(bit_buffer_get_size_bytes(instance->rx_buffer) != 1) {
+            FURI_LOG_D(TAG, "Unexpected Select response size");
+            instance->state = St25tbPollerStateActivationFailed;
+            ret = St25tbErrorCommunication;
+            break;
+        }
+
+        if(bit_buffer_get_byte(instance->rx_buffer, 0) != data->chip_id) {
+            FURI_LOG_D(TAG, "ChipID mismatch");
+            instance->state = St25tbPollerStateActivationFailed;
+            ret = St25tbErrorColResFailed;
+            break;
+        }
+        instance->state = St25tbPollerStateActivated;
+
+        ret = st25tb_poller_async_get_uid(instance, data->uid);
+        if(ret != St25tbErrorNone) {
+            instance->state = St25tbPollerStateActivationFailed;
+            break;
+        }
+        data->type = st25tb_get_type_from_uid(data->uid);
+
+        bool read_blocks = true;
+        for(uint8_t i = 0; i < st25tb_get_block_count(data->type); i++) {
+            ret = st25tb_poller_async_read_block(instance, &data->blocks[i], i);
+            if(ret != St25tbErrorNone) {
+                read_blocks = false;
+                break;
+            }
+        }
+        if(!read_blocks) {
+            break;
+        }
+        ret = st25tb_poller_async_read_block(
+            instance, &data->system_otp_block, ST25TB_SYSTEM_OTP_BLOCK);
+        if(ret != St25tbErrorNone) {
+            read_blocks = false;
```

Comment:
doesn't matter

Reply:
Yeah that's useless, I left that behind while refactoring, thanks again :)